tasks/package.yaml

```diff
@@ -9,9 +9,9 @@
     - sudo_pkg
 
 - name: Install sudo for RedHat
-  yum:
-    name: sudo
-    state: present
+  action:
+    "{{ ansible_pkg_mgr }} pkg=sudo
+    state=present"
```

Comment:
It works only because you are using the one-line Ansible params format. But normally, you don't have to quote it at all. Anyway, I prefer pure YAML instead.